modules/dailymotionBidAdapter.js

```diff
@@ -130,76 +130,96 @@ export const spec = {
    * @param {BidderRequest} bidderRequest
    * @return ServerRequest Info describing the request to the server.
    */
-  buildRequests: (validBidRequests = [], bidderRequest) => validBidRequests.map(bid => ({
-    method: 'POST',
-    url: 'https://pb.dmxleo.com',
-    data: {
-      pbv: '$prebid.version$',
-      bidder_request: {
-        gdprConsent: {
-          apiVersion: deepAccess(bidderRequest, 'gdprConsent.apiVersion', 1),
-          consentString: deepAccess(bidderRequest, 'gdprConsent.consentString', ''),
-          // Cast boolean in any case (eg: if value is int) to ensure type
-          gdprApplies: !!deepAccess(bidderRequest, 'gdprConsent.gdprApplies'),
-        },
-        refererInfo: {
-          page: deepAccess(bidderRequest, 'refererInfo.page', ''),
+  buildRequests: function(validBidRequests = [], bidderRequest) {
+    // check consent to be able to read user cookie
+    const allowCookieReading =
+      // No GDPR applies
+      !deepAccess(bidderRequest, 'gdprConsent.gdprApplies') ||
+      // OR GDPR applies and we have global consent
+      deepAccess(bidderRequest, 'gdprConsent.vendorData.hasGlobalConsent') === true ||
+      (
+        deepAccess(bidderRequest, 'gdprConsent.vendorData.vendor.consents.573') === true &&
+        deepAccess(bidderRequest, 'gdprConsent.vendorData.purpose.consents.1') === true &&
```

Comment:
surely there is a more compact way to code this

Reply:
maybe something like this

`const li = 'legitimateInterests';
const c = 'consents';

const hasConsent = (type, path) => deepAccess(bidderRequest.gdprConsent.vendorData.purpose[type], path) === true;

const allowCookieReading = 
  // No GDPR applies
  !deepAccess(bidderRequest, 'gdprConsent.gdprApplies') ||
  // OR GDPR applies and we have global consent
  deepAccess(bidderRequest, 'gdprConsent.vendorData.hasGlobalConsent') === true ||
  // OR specific consents are given
  (
    deepAccess(bidderRequest, 'gdprConsent.vendorData.vendor.consents.573') === true &&
    hasConsent(c, '1') &&
    hasConsent(c, '3') &&
    hasConsent(c, '4') &&
    // emulate flexible purpose by checking if the default consent or legitimate interest is set
    (hasConsent(c, '2') || hasConsent(li, '2')) &&
    (hasConsent(c, '7') || hasConsent(li, '7')) &&
    (hasConsent(c, '9') || hasConsent(li, '9')) &&
    (hasConsent(c, '10') || hasConsent(li, '10'))
  );`

Reply:
Done! Let me know what you think about this new proposal